lib/dry/cli/styles.rb

```diff
@@ -0,0 +1,187 @@
+      #
+      # @since 1.3.0
+      def stylize(text)
+        StyledText.new(text)
+      end
+
+      # Styled text
+      #
+      # @since 1.3.0
+      class StyledText
+        def initialize(text)
+          @text = text
+        end
+
+        # Makes `StyledText` printable
+        #
+        # @since 1.3.0
+        def to_s
+          @text
+        end
+
+        # since 1.3.0
+        def bold
+          chainable_update!("\e[#{BOLD}m#{@text}")
+        end
+
+        # since 1.3.0
+        def dim
+          chainable_update!("\e[#{DIM}m#{@text}")
+        end
+
+        # since 1.3.0
+        def italic
+          chainable_update!("\e[#{ITALIC}m#{@text}")
+        end
+
+        # since 1.3.0
+        def underline
+          chainable_update!("\e[#{UNDERLINE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def blink
+          chainable_update!("\e[#{BLINK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def reverse
+          chainable_update!("\e[#{REVERSE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def invisible
+          chainable_update!("\e[#{INVISIBLE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def black
+          chainable_update!("\e[#{BLACK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def red
+          chainable_update!("\e[#{RED}m#{@text}")
+        end
+
+        # since 1.3.0
+        def green
+          chainable_update!("\e[#{GREEN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def yellow
+          chainable_update!("\e[#{YELLOW}m#{@text}")
+        end
+
+        # since 1.3.0
+        def blue
+          chainable_update!("\e[#{BLUE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def magenta
+          chainable_update!("\e[#{MAGENTA}m#{@text}")
+        end
+
+        # since 1.3.0
+        def cyan
+          chainable_update!("\e[#{CYAN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def white
+          chainable_update!("\e[#{WHITE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_black
+          chainable_update!("\e[#{ON_BLACK}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_red
+          chainable_update!("\e[#{ON_RED}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_green
+          chainable_update!("\e[#{ON_GREEN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_yellow
+          chainable_update!("\e[#{ON_YELLOW}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_blue
+          chainable_update!("\e[#{ON_BLUE}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_magenta
+          chainable_update!("\e[#{ON_MAGENTA}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_cyan
+          chainable_update!("\e[#{ON_CYAN}m#{@text}")
+        end
+
+        # since 1.3.0
+        def on_white
+          chainable_update!("\e[#{ON_WHITE}m#{@text}")
+        end
+
+        private
+
```

Comment:
Can we use an attr_reader throughout, instead of the instance variable?

```suggestion
       attr_reader :text
```

Then change almost all `@text`s to just `text`